### PR TITLE
core/chains/evm/client: eth_call: include duplicate legacy field for compatability

### PR DESCRIPTION
**Pull request by jmank88**

### CODE REVIEW

File: core/chains/evm/client/rpc_client.go [hunk 1]
```diff
@@ -817,7 +821,51 @@ func (r *rpcClient) PendingCallContract(ctx context.Context, msg interface{}) (v
...
+// COPIED FROM go-ethereum/ethclient/gethclient - must be kept up to date!
+// Modified to include legacy 'data' as well as 'input' in order to support non-compliant servers.
+func toCallArg(msg ethereum.CallMsg) interface{} {
+	arg := map[string]interface{}{
+		"from": msg.From,
+		"to":   msg.To,
+	}
+	if len(msg.Data) > 0 {
+		arg["input"] = hexutil.Bytes(msg.Data)
+		arg["data"] = hexutil.Bytes(msg.Data) // duplicate legacy field for compatability
```
Comment (makramkd) [inline core/chains/evm/client/rpc_client.go:851]: This is interesting - so geth included these before and has now removed it in the latest version. Is that correct?

Comment (makramkd) [reply core/chains/evm/client/rpc_client.go:851]: This is indeed nonstandard: https://ethereum.org/en/developers/docs/apis/json-rpc#eth_call

File: core/chains/evm/client/rpc_client.go [hunk 2]
```diff
@@ -817,7 +821,51 @@ func (r *rpcClient) PendingCallContract(ctx context.Context, msg interface{}) (v
...
+	}
+	// It's negative.
+	if number.IsInt64() {
+		return rpc.BlockNumber(number.Int64()).String()
+	}
+	// It's negative and large, which is invalid.
+	return fmt.Sprintf("<invalid %d>", number)
+}
 
+// COPIED FROM go-ethereum/ethclient/gethclient - must be kept up to date!
```
Comment (makramkd) [inline core/chains/evm/client/rpc_client.go:842]: What commit sha of go-ethereum was this copied from?

Comment (jmank88) [reply core/chains/evm/client/rpc_client.go:842]: The one that is imported in the go.mod

File: core/chains/evm/client/rpc_client.go [hunk 3]
```diff
@@ -803,21 +810,69 @@ func (r *rpcClient) PendingCallContract(ctx context.Context, msg interface{}) (v
...
+	if number.IsInt64() {
+		return rpc.BlockNumber(number.Int64()).String()
+	}
+	// It's negative and large, which is invalid.
+	return fmt.Sprintf("<invalid %d>", number)
+}
+
+// COPIED FROM go-ethereum/ethclient/gethclient - must be kept up to date!
+// Modified to include legacy 'data' as well as 'input' in order to support non-compliant servers.
+func toCallArg(msg ethereum.CallMsg) interface{} {
```
Comment (stackman27) [inline core/chains/evm/client/rpc_client.go:851]: @jmank88 ethereum recently updated this function to and added extra fields, should we reflect those changes?

https://github.com/ethereum/go-ethereum/blob/master/ethclient/ethclient.go#L642

**Review by stackman27 — DISMISSED** [review]
LGTM

File: core/chains/evm/client/rpc_client.go (hunk 3 above)
Comment (stackman27) [reply core/chains/evm/client/rpc_client.go:851]: nvm 13.1.8 doesnot use it,  just need to ensure we update this function if we do modify to new geth version

Comment (jmank88) [reply core/chains/evm/client/rpc_client.go:851]: I copied from the version we import to minimize changes

**Review by stackman27 — APPROVED** [review]

**Pull request merged**